cli/azd/pkg/infra/provisioning/manager.go

```diff
@@ -86,12 +92,59 @@ func (m *Manager) Deploy(ctx context.Context) (*DeployResult, error) {
 		return nil, fmt.Errorf("updating environment with deployment outputs: %w", err)
 	}
 
+	stAccount, hasBindMountOperation := m.env.LookupEnv(operationsBindMount)
+	if hasBindMountOperation {
+		m.console.ShowSpinner(ctx, "uploading files for binding mount", input.StepFailed)
+		for key, value := range m.env.Dotenv() {
```

Comment:
Like @weikanglim was, I'm a little nervous about storing this control data in the `.env` file and using it from there.  I would almost prefer that we write a new file into the `infra` folder or something that tells `azd` what content it needs to upload for bind mounts.  Like if we also had a `<aspire-resource-name>.bindMounts.json` file in the `infra` folder that contained something like this:

```json
[{ "source": <source-path-relative-to-app-host>?, "account": <account-name>, "share": <share-name> }]
```

and during `deploy`, if this file exists, we follow the directions in it.

That feels a little more first class and IaC'y.

Reply:
I can do that.
The general concept is built-in actions for azd. Which could be for after provision or before deploy.

In this case, I'm still using the end of provision and re-using the output of resources like:

![image](https://github.com/Azure/azure-dev/assets/24213737/6059e3c6-0db6-4742-aa93-578ef20dbdb7)

Note:  The image is out-of-date. I updated the text to `Done: Mount Volume: <path>`

Reply:
Updated to use `azd.operations.yaml` as an extension of the IaC and outside the .env